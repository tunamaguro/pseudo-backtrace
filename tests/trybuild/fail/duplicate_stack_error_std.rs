use pseudo_backtrace::StackError;

#[derive(Debug, StackError)]
struct DupStdAttr {
    #[stack_error(std)]
    #[stack_error(std)]
    source: std::io::Error,
    #[location]
    location: &'static core::panic::Location<'static>,
}

impl core::fmt::Display for DupStdAttr {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
        "dup".fmt(f)
    }
}
impl core::error::Error for DupStdAttr {}

fn main() {}
