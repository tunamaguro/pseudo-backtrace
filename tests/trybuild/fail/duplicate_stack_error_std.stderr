error: duplicate `#[stack_error(...)]` attribute
 --> tests/trybuild/fail/duplicate_stack_error_std.rs:6:5
  |
6 |     #[stack_error(std)]
  |     ^^^^^^^^^^^^^^^^^^^
